src/NServiceBus.Gateway.Tests/APIApprovals.Approve.approved.txt

```diff
@@ -130,6 +130,13 @@ namespace NServiceBus.Gateway.Channels
         public string Type { get; set; }
     }
 }
+namespace NServiceBus.Gateway.Recoverability
+{
+    public class static IncomingMessageExtensions
+    {
+        public static int GetCurrentRetry(this NServiceBus.Transports.IncomingMessage message) { }
```

Comment:
Does it make sense for the gateway to expose this extension method? Alternatively we pass in the retry number and the exception details into the retry policy.